src/plugins/telemetry/server/plugin.ts

```diff
@@ -134,17 +129,26 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
 
     // If the opt-in selection cannot be changed, set it as early as possible.
     const { optIn, allowChangingOptInStatus } = this.initialConfig;
-    if (allowChangingOptInStatus === false) {
-      this.isOptedIn$.next(optIn);
-    }
+    const initialStatus = allowChangingOptInStatus === false ? optIn : undefined;
+
+    // Poll for the opt-in status
+    this.isOptedIn$ = timer(0, OPT_IN_POLL_INTERVAL_MS).pipe(
+      takeUntil(this.pluginStop$),
+      exhaustMap(() => this.getOptInStatus()),
+      startWith(initialStatus),
+      distinctUntilChanged(),
+      filter((isOptedIn): isOptedIn is boolean => typeof isOptedIn === 'boolean'),
+      tap((optedIn) => (this.isOptedIn = optedIn)),
+      shareReplay(1)
+    );
   }
 
   public setup(
     { analytics, http, savedObjects }: CoreSetup,
     { usageCollection, telemetryCollectionManager }: TelemetryPluginsDepsSetup
   ): TelemetryPluginSetup {
-    if (this.isOptedIn$.value !== undefined) {
-      analytics.optIn({ global: { enabled: this.isOptedIn$.value } });
+    if (this.isOptedIn !== undefined) {
```

Comment:
I wonder if this is ever true: we are populating it in the pipe of `this.isOptedIn$`, but that observable is only subscribed in the `start` phase. AFAIUnderstand, observables are not evaluated until they have a subscription, do they?